src/burnchains/burnchain.rs

```diff
@@ -546,6 +543,56 @@ impl Burnchain {
         ret
     }
 
+    /// Verify that two or more block commits do not consume the same VRF key.
+    /// If a key is consumed more than once, then pick the block-commit with the highest burn (to
+    /// stop griefing attacks).  In case of ties, pick the block-commit that occurs earlier in the
+    /// block.
+    fn filter_block_commits_with_same_VRF_key(mut checked_ops: Vec<BlockstackOperationType>) -> Vec<BlockstackOperationType> {
+        debug!("Check Blockstack transactions: filter commits that consume the same VRF key");
+        assert!(Burnchain::ops_are_sorted(&checked_ops));
+
+        let mut ret = Vec::with_capacity(checked_ops.len());
+
+        let mut collisions : HashMap<(u64, u32), BlockstackOperationType> = HashMap::new();
+        for op in checked_ops.drain(..) {
```

Comment:
I'm guilty of this a lot too -- but many places where we `.drain(..)`, we can more appropriately use `.into_iter()`

Reply:
sure, can fix